src/fontra/client/web-components/ui-form.js

```diff
@@ -181,6 +183,11 @@ export class Form extends SimpleElement {
     };
     inputElement.onchange = (event) => {
       let value = parseFloat(inputElement.value);
+      if (isNaN(value)) {
```

Comment:
Let's move this bit to just before the `this._lastValidFieldValues[fieldItem.key] = value;` line, that way it is more similar to the rotation case.

Reply:
Sorry, but I don't know where you want to move it.

Reply:
Isn't it pretty much the same? `if (isNaN(value)) {` is right after `let value = parseFloat(inputElement.value);` in both cases.

Reply:
Rereading the code: never mind, the current place is best after all.

(I was hoping the code would look more similar to the rotation case, but I think it indeed needs to be before the "reportValidity" section.

Reply:
'Code Design' ;)